### DISCUSSION TITLE
Have clients in a client group visited by same vehicle?

### DISCUSSION THREAD

**Discussion by imayao** [PyVRP/PyVRP#718] [Q&A]
Does "client_group" mean that customers within a single group are visited by the same vehicle, or on the same route, while different groups cannot be visited by the same vehicle at the same time?
I am working on a VRP problem, trying to arrange for customers in some small towns to be served by the same vehicle, instead of having three or four different vehicles visit the same town.

**Comment by N-Wouda** [comment]
Hi @imayao, this is called a clustered VRP (possibly with weak cluster links; see #470 for additional details). We do not yet support this in full generality, although it is something we plan to pick up sometime next year. So, to answer the question:

> Does "client_group" mean that customers within a single group are visited by the same vehicle, or on the same route, while different groups cannot be visited by the same vehicle at the same time?

Not yet, but it should be possible sometime soon-ish. As a workaround, you could for now make the arcs into (and leaving) each town quite expensive by setting an artificially high distance on them. That would motivate the solver to only visit each town once, using a single route/vehicle.

**Reply by imayao** [reply]
Thanks for your help！
The method you mentioned is very interesting and i will give it a try. However, I'm concerned that altering the distances in this way might have the side effect of reducing the number of customers that could originally be served. 
My VRP issue is a CVRP problem without time windows, and even after decomposition, I still have to deal with about 1000 customers. For such a large scale with uneven urban and rural distribution, it is really an unbelievably tough optimization.

**Reply by N-Wouda** [reply]
@imayao no problem, let us know how it goes! 🤞
